lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -44,6 +45,11 @@
         description:
             - Name of the resource group containing the virtual machine.
         required: true
+    resource:
```

Comment:
Even though the SDK calls this `resource`, I'm a little loathe to name the module arg something that so closely mirrors `resource_group` - seems like a trip hazard. I've seen some other Azure tooling call it `environment` - while that collides with our existing `environment` play/task keyword, I'd argue it's still less confusing. Open to ideas here...